File: WcfChat/Common/IChat.cs
using System.ServiceModel;

namespace Common
{
	[ServiceContract(CallbackContract = typeof(IChatCallback))]
	public interface IChat
	{
		[OperationContract(IsOneWay = true)]
		void Login(string name);

		[OperationContract(IsOneWay = true)]
		void Logout();

		[OperationContract(IsOneWay = true)]
		void Send(string to, string message);
	}
}

File: rmi-chat/src/common/IChat.java
package common;

import java.rmi.Remote;
import java.rmi.RemoteException;

public interface IChat extends Remote {

	void login(String name, IChatCallback callback) throws RemoteException;
	
	void logout(IChatCallback callback) throws RemoteException;
	
	void send(String to, String message, IChatCallback callback) throws RemoteException;
}
